components/server/src/server.ts

```diff
@@ -344,12 +349,14 @@ export class Server {
     }
 
     public async stop() {
-        await this.debugApp.stop();
-        await this.stopServer(this.iamSessionAppServer);
-        await this.stopServer(this.monitoringHttpServer);
-        await this.stopServer(this.httpServer);
-        await this.stopServer(this.apiServer);
-        this.disposables.dispose();
+        await Promise.allSettled([
+            this.debugApp.stop(),
+            this.stopServer(this.iamSessionAppServer),
+            this.stopServer(this.monitoringHttpServer),
+            this.stopServer(this.httpServer),
+            this.stopServer(this.apiServer),
+            new Promise(() => this.disposables.dispose()),
+        ]).catch((err) => log.error("error while stopping server", { err }));
```

Comment:
As I was working on a similar fix I build something that gives us a signal (when stopping doesn't succeeed in 30s). Some sort of signal (metrics or log) would be good to know if we again don't stop gracefully.

Leaving it here:

```
        // run each stop with a timeout of 30s
        async function race(workLoad: Promise<any>, task: string, ms: number = 30 * 1000): Promise<void> {
            const before = Date.now();
            let timedOut = false;
            const timeout = new Promise<void>((resolve) =>
                setTimeout(() => {
                    timedOut = true;
                    resolve();
                }, ms),
            );
            await Promise.race([workLoad, timeout]).then(() => {
                const duration = Date.now() - before;
                if (timedOut) {
                    log.warn(`task ${task} timed out after ${duration}ms`);
                } else {
                    log.info(`task ${task} took ${duration}ms`);
                }
            });
        }
        await race(
            Promise.allSettled([
                race(this.debugApp.stop(), "debugapp"),
                race(this.stopServer(this.iamSessionAppServer), "stop iamsessionapp"),
                race(this.stopServer(this.monitoringHttpServer), "stop monitoringapp"),
                race(this.stopServer(this.httpServer), "stop httpserver"),
                race(this.stopServer(this.apiServer), "stop api server"),
                race((async () => this.disposables.dispose())(), "dispose disposables"),
            ]),
            "all stopping",
        );

        
```

Reply:
@svenefftinge I opened https://github.com/gitpod-io/gitpod/pull/18548 to extract these bits. Does it make sense to have another PR? WOuld you open another PR? cc @geropl since you are reviewing

Reply:
Oh, I thought this PR was on hold. So opted for raising one for the was termination https://github.com/gitpod-io/gitpod/pull/18550

Reply:
For the signal though I'm not sure we need race, missing message should be enough to indicate?